include/executor/engine/binary_numeric.ipp

```diff
@@ -157,9 +157,22 @@ TypeF<T> Executor::runMinOp(ValVariant &Val1, const ValVariant &Val2) const {
   T &Z1 = Val1.get<T>();
   const T &Z2 = Val2.get<T>();
   const T kZero = 0.0;
-  // TODO: canonical and arithmetical NaN
-  if (std::isnan(Z2)) {
-    Z1 = Z2;
+  if (std::isnan(Z1) || std::isnan(Z2)) {
+    if (std::isnan(Z2)) {
+      Z1 = Z2;
+    }
+    // Set the most significant bit of the payload to 1.
+    if (sizeof(T) == sizeof(uint32_t)) {
+      uint32_t I32;
+      std::memcpy(&I32, &Z1, sizeof(T));
+      I32 |= (uint32_t)0x1 << 22;
+      std::memcpy(&Z1, &I32, sizeof(T));
```

Comment:
`reinterpret_cast<uint32_t &>(Z1) |= (static_cast<uint32_t>(0x01U) << 22)`;

Reply:
It looks like some CI tests failed because of `reinterpret_cast`. Do you have any suggestions?

Reply:
I have tried `*reinterpret_cast<uint32_t *>`, but it's the same.